enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestionResult/ResultTitle.tsx

```diff
@@ -0,0 +1,74 @@
+import { type ReactNode, isValidElement } from "react";
+
+import { Anchor, Stack, Text } from "metabase/ui";
+
+import { useInteractiveQuestionContext } from "../InteractiveQuestion/context";
+import { getQuestionTitle } from "../QuestionTitle";
+
+import type { InteractiveQuestionResultProps } from "./InteractiveQuestionResult";
+
+interface ResultTitleTextProps
+  extends Pick<InteractiveQuestionResultProps, "withResetButton"> {
+  text: string | ReactNode;
+  isQuestionChanged?: boolean;
+  onReset?: () => void;
+  originalName?: string | null;
+}
+
+const ResultTitleText = ({
+  text,
+  withResetButton = false,
+  isQuestionChanged = false,
+  onReset,
+  originalName,
+}: ResultTitleTextProps) => (
+  <Stack spacing="xs">
+    {originalName && withResetButton && isQuestionChanged && (
+      <Text fw={600} size="sm">
+        Return to
+        <Anchor size="sm" ml="xs" color="brand" onClick={onReset}>
+          {originalName}
+        </Anchor>
+      </Text>
+    )}
+    <Text fw={700} c="var(--mb-color-text-primary)" fz="xl">
```

Comment:
I recall [AlexP mentioned on Slack](https://metaboat.slack.com/archives/C505ZNNH4/p1730220078470559) that we should use existing color names for inline Mantine props for now, sadly we don't have `c="text-primary"` yet

Reply:
Yeah, I saw this too. I just assumed that I would leave it with the `var` for now